leaderelection/context.go

```diff
@@ -181,6 +206,25 @@ func standardBucketName(ordinal uint32, queueName string, cc ComponentConfig) st
 	return strings.ToLower(fmt.Sprintf("%s.%s.%02d-of-%02d", cc.Component, queueName, ordinal, cc.Buckets))
 }
 
+// NewEndpointsBucketSet returns a hash.BucketSet consists of the buckets based on
+// the given ComponentConfig.
+func NewEndpointsBucketSet(cc ComponentConfig) *hash.BucketSet {
+	names := make(sets.String, cc.Buckets)
+	for i := uint32(0); i < cc.Buckets; i++ {
+		names.Insert(endpointsBucketName(i, cc))
+	}
```

Comment:
Do we need to create services or we'll do that elsewhere?

Reply:
Yeah, autoscaler needs to watch the changes and create the service.